src/transformers/trainer.py

```diff
@@ -4276,6 +4287,38 @@ def push_to_hub(
             ignore_patterns=["_*", f"{PREFIX_CHECKPOINT_DIR}-*"],
         )
 
+        if gguf_config is not None:
+            if not isinstance(gguf_config, GgufConfig):
+                raise ValueError("Please pass an instance of `GgufConfig`")
+
+            from gradio_client import Client
+            from huggingface_hub.utils import get_token_to_send
+
+            if get_token_to_send(token) is None:
+                logger.warning(
+                    "Make sure to login with `huggingface-cli login` before running the GGUF conversion, will silently ignore the GGUF conversion."
+                )
+                return upload_results
+
+            if gguf_config.duplicate_space:
+                client = Client.duplicate(gguf_config.space_name, hf_token=get_token_to_send(token))
+            else:
+                client = Client(gguf_config.space_name)
+
+            job = client.submit(
+                model_id=self.hub_model_id,
+                q_method=gguf_config.quantization_method,
+                private_repo=gguf_config.private,
+                token=get_token_to_send(token),
+                api_name="/predict",
+            )
+
+            logger.info(
+                f"GGUF conversion Space is running with the status {job.status().code} - please monitor your HF profile to see if the converted model has been pushed"
+            )
```

Comment:
hmmmm.... I'm really not sure about this, my gut feeling is that it's pretty dirty piggy-backing gradio here. 

I'd like for this general logic to be moved out to a function rather than being injected directly into a core trainer method

Reply:
Makes sense, will move that out in a separate method !